cleanup.go

```diff
@@ -16,21 +15,21 @@ type terminateOptions struct {
 }
 
 // TerminateOption is a type that represents an option for terminating a container.
-type TerminateOption func(*terminateOptions)
+type TerminateOption func(*DockerContainer)
```

Comment:
yeah, i had similar feeling, but wasn't sure if shall i introduce a separate time, when i was opening this PR i tried to make as minimal breaking change but seems it has to be. open for suggestions 

i like the idea of `maintain the dedicated type but export it and the fields`

Reply:
I initially thought the dedicated type would work, but that wont ensure that all Container implementations support the options, which I'm not a fan of as a consumer would be left wondering why their options didn't work.

Reply:
the other thought i had before i come to this was introducing it in the `ContainerRequest`  and add getter in the `container` interface

Reply:
I thought about setters too, but I think that would make it much harder to discover for users.

I can't think of any perfect solution ATM, something mull over.

Reply:
i have made a change and revert it to make sure it uses `terminateOptions`  instead of the dependency on DockerContainer type. 

also added more test to show case the usage and the changes of configuration 

defaults of volumes is nil, timeout is 10   and context is passed already on `Terminate()` , lmk wdyt ?